src/bootstrap/compile.rs

```diff
@@ -1137,10 +1136,13 @@ pub fn run_cargo(builder: &Builder,
     // contents (the list of files to copy) is different or if any dep's mtime
     // is newer then we rewrite the stamp file.
     deps.sort();
-    let mut stamp_contents = Vec::new();
-    if let Ok(mut f) = File::open(stamp) {
-        t!(f.read_to_end(&mut stamp_contents));
-    }
+    let stamp_contents = match fs::read(stamp) {
+        Ok(contents) => contents,
+        Err(ref err) if err.kind() == ErrorKind::InvalidData => {
+            panic!("could not read contents of stamp");
```

Comment:
Why would InvalidData come up? I could maybe see it if we were reading to string, but with just a Vec<u8> return value I don't quite see it -- and this does appear to be new code

Reply:
I was trying to keep the behavior that errors opening the file are ignored but errors reading the file panic. You're right that InvalidData won't come up when we're reading bytes -- it's probably best to just do something like `let contents = fs::read(stamp).ok();`?